sql/api/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -143,7 +142,7 @@ final class Decimal extends Ordered[Decimal] with Serializable {
       // the scale is 2. The expected precision should be 2.
       this._precision = decimal.scale
       this._scale = decimal.scale
-    } else if (decimal.scale < 0 && !SQLConf.get.allowNegativeScaleOfDecimalEnabled) {
+    } else if (decimal.scale < 0 && !SqlApiConf.get.allowNegativeScaleOfDecimalEnabled) {
```

Comment:
how do we bind `SqlApiConf.get` to the active SQLConf at the server side?

Reply:
As soon as SQLConf is loaded we connect them: https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L185